pyani_plus/tools.py

```diff
@@ -289,3 +289,31 @@ def get_show_diff(cmd: str | Path = "show-diff") -> ExternalToolData:
         raise RuntimeError(msg)
 
     return ExternalToolData(exe_path, version)
+
+
+def get_dnadiff(cmd: str | Path = "dnadiff") -> ExternalToolData:
+    """Return dnadiff and version as a named tuple.
+
+    We expect the tool to behave as follows:
+
+    .. code-block:: bash
+
+        $ dnadiff -V
+            dnadiff
+            DNAdiff version 1.3
+
+    Here the function would return the binary path and "1.3" as the version.
+
+    Raises a RuntimeError if the command cannot be found, run, or if the
+    version cannot be inferred - this likely indicates a dramatically
+    different version of the tool with different behaviour.
+    """
+    exe_path, output = _get_path_and_version_output(cmd, ["-V"])
+
+    match = re.search(r"(?<=version\s)[0-9\.]*", output)
+    version = match.group().strip() if match else None
+    if not version:
+        msg = f"Executable exists at {exe_path} but could not retrieve version"
+        raise RuntimeError(msg)
+
+    return ExternalToolData(exe_path, version)
```

Comment:
Can you add something to ``test_tools.py``, both a good and a bad case using the existing mock commands?

Reply:
Sure! I just realised I haven't tested this part of the code!